app/controllers/public_health_controller.rb

```diff
@@ -493,161 +440,62 @@ def advanced_filter(patients, filters)
-        when 'weeks'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(last_date_of_exposure) <= ?', number.weeks.from_now).where('Date(last_date_of_exposure) >= ?', Date.today)
-        when 'months'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(last_date_of_exposure) <= ?', number.months.from_now).where('Date(last_date_of_exposure) >= ?', Date.today)
-        end
-      end
+    if timeframe[:before].present?
+      # probabbly need to adjust for local timezone?
+      before = timeframe[:before]
+      patients = patients.where('created_at <= ?', before) if field == :created_at
+      patients = patients.where('latest_assessment_at <= ?', before) if field == :latest_assessment_at
+      patients = patients.where('last_date_of_exposure <= ?', before) if field == :last_date_of_exposure
+      patients = patients.where('symptom_onset <= ?', before) if field == :symptom_onset
     end
+
     patients
   end
 
-  def advanced_filter_symptom_onset_relative(patients, filter)
-    case filter[:relativeOption]
-    when 'today'
-      patients = patients.where('Date(symptom_onset) = ?', Date.today)
-    when 'tomorrow'
-      patients = patients.where('Date(symptom_onset) = ?', Date.tomorrow)
-    when 'yesterday'
-      patients = patients.where('Date(symptom_onset) = ?', Date.yesterday)
-    when 'custom'
-      case filter[:value][:when]
-      when 'past'
-        case filter[:value][:unit]
-        when 'days'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(symptom_onset) >= ?', number.days.ago).where('Date(symptom_onset) <= ?', Date.today)
-        when 'weeks'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(symptom_onset) >= ?', number.weeks.ago).where('Date(symptom_onset) <= ?', Date.today)
-        when 'months'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(symptom_onset) >= ?', number.months.ago).where('Date(symptom_onset) <= ?', Date.today)
-        end
-      when 'next'
-        case filter[:value][:unit]
-        when 'days'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(symptom_onset) <= ?', number.days.from_now).where('Date(symptom_onset) >= ?', Date.today)
-        when 'weeks'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(symptom_onset) <= ?', number.weeks.from_now).where('Date(symptom_onset) >= ?', Date.today)
-        when 'months'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(symptom_onset) <= ?', number.months.from_now).where('Date(symptom_onset) >= ?', Date.today)
-        end
-      end
+  def advanced_filter_relative_date(patients, field, filter)
+    return patients unless filter.dig(:value, :tzOffset).present?
```

Comment:
iif I comment this line out, I do see the filters work.  I realize that's not the solution but could help with debugging

Reply:
everything seems to work for me now (not sure why). just need to find a way to test the timezone stuff